xnmt/encoder.py

```diff
@@ -20,19 +23,37 @@ class Encoder(HierarchicalModel):
   """
   An Encoder is a class that takes an ExpressionSequence as input and outputs another encoded ExpressionSequence.
   """
-  def transduce(self, sent):
+  def transduce(self, embed_sent):
     """Encode inputs representing a sequence of continuous vectors into outputs that also represent a sequence of continuous vectors.
 
     :param sent: The input to be encoded. In the great majority of cases this will be an ExpressionSequence.
       It can be something else if the encoder is over something that is not a sequence of vectors though.
     :returns: The encoded output. In the great majority of cases this will be an ExpressionSequence.
       It can be something else if the encoder is over something that is not a sequence of vectors though.
     """
-    raise NotImplementedError('transduce must be implemented in Encoder subclasses')
+    raise NotImplementedError('Unimplemented transduce for class:', self.__class__.__name__)
+
+  def calc_reinforce_loss(self, reward):
+    return None
 
 class BuilderEncoder(Encoder):
-  def transduce(self, sent):
-    return embedder.ExpressionSequence(expr_list=self.builder.transduce(sent))
+  def transduce(self, embed_sent):
+    out = None
+    if hasattr(self.builder, "transduce"):
+      out = self.builder.transduce(embed_sent.get())
+    elif hasattr(self.builder, "initial_state"):
+      out = self.builder.initial_state().transduce(embed_sent.get())
+    else:
+      raise NotImplementedError("Unimplemented transduce logic for class:",
+                                self.builder.__class__.__name__)
+
+    return ExpressionSequence(expr_list=out)
+
+class IdentityEncoder(Encoder, Serializable):
+  yaml_tag = u'!IdentityEncoder'
+
+  def transduce(self, embed_sent):
+    return ExpressionSequence(expr_list = sent.get())
```

Comment:
This code should just return `embed_sent`, right? Encoders assume they receive an expression sequence and return an expression sequence.